contracts/test/FeeContract.t.sol

```diff
@@ -0,0 +1,278 @@
+    }
+
+    function testFuzz_nonExistentFunction() public payable {
+        address fcAddress = address(feeContract);
+        (bool success,) =
+            fcAddress.call{ value: msg.value }(abi.encodeWithSignature("withdraw(address)", "0x"));
+
+        //assert that the transaction was not successful
+        assertFalse(success);
+
+        //assert that the balance of the fee contract is still zero
+        assertEq(address(feeContract).balance, 0);
+    }
+
+    //test deposits with a large amount reverts
+    function test_depositMaxAmount() public {
+        address user = makeAddr("user");
+        uint256 amount = feeContract.MAX_DEPOSIT_AMOUNT() + 1;
+
+        vm.expectRevert(FeeContract.DepositTooLarge.selector);
+
+        //deposit for the user
+        feeContract.deposit{ value: amount }(user);
+    }
+
+    //test deposits with a less than the min amount reverts
+    function test_depositMinAmount() public {
+        address user = makeAddr("user");
+        uint256 amount = feeContract.MIN_DEPOSIT_AMOUNT() - 0.0001 ether;
+
+        vm.expectRevert(FeeContract.DepositTooSmall.selector);
+
+        //deposit for the user
+        feeContract.deposit{ value: amount }(user);
+    }
+
+    //test deposits with invalid user address reverts
+    function test_invalidUserAddress() public {
+        address user = address(0);
+        uint256 amount = 0.5 ether;
+
+        vm.expectRevert(FeeContract.InvalidUserAddress.selector);
+
+        //deposit for the user
+        feeContract.deposit{ value: amount }(user);
+    }
+
+    function testFuzz_depositEvent(address user, uint256 amount) public payable {
+        vm.assume(user != address(0));
+        amount = bound(amount, feeContract.MIN_DEPOSIT_AMOUNT(), feeContract.MAX_DEPOSIT_AMOUNT());
+
+        uint256 balanceBeforeUser = feeContract.getBalance(user);
+
+        vm.expectEmit(true, false, false, true);
+        // We emit the event we expect to see.
+        emit FeeContract.Deposit(user, amount);
+
+        //deposit for the two users
+        feeContract.deposit{ value: amount }(user);
+
+        //get the balance for that user after the deposit
+        uint256 balanceAfterUser = feeContract.getBalance(user);
+
+        //test that the users' balances have been incremented accurately
+        assertEq(balanceAfterUser, balanceBeforeUser + amount);
+
+        //test that the smart contract has the accumulative balance for both users
+        assertEq(address(feeContract).balance, amount);
+    }
+
+    function testFuzz_newUserHasZeroBalance(address user) public {
+        vm.assume(user != address(0));
+
+        uint256 balance = feeContract.getBalance(user);
+
+        assertEq(balance, 0);
+    }
+}
+
+contract FeeContractUpgradabilityTest is Test {
+    address payable public proxy;
+    address public admin;
+    FeeContract public feeContractProxy;
+    DeployFeeContract public deployer = new DeployFeeContract();
```

Comment:
if we set the `admin` to what's derived from env.mnemonic, then we wouldn't need this deployer contract variable, just storing the `address proxy` is sufficient.

Reply:
true, there is no need for the deployer to be a state variable https://github.com/EspressoSystems/espresso-sequencer/commit/3c7c6fe7fd90130907e29bd90863354a7cd58bea